ar71xx: add support for Mikrotik RB SXT 2nD r3

zorxd opened the pull request:
The device can be added to the white list since it boots fine with the generated NOR flash image, as long as the flash is erased before the sysupgrade.
The board id is 911L. Machine name is MikroTik RouterBOARD SXT 2nD r3.

zorxd commented:
anything wrong with this PR?

ynezz commented:
> anything wrong with this PR?

Kind of, it's ar71xx which is more or less in maintenance mode (fixes only) and new development should happen only in ath79 (DTS based ar71xx successor). But since you've this PR pending here  since October and it seems trivial enough, I'll make exception and merge it if you could incorporate following (from https://openwrt.org/submitting-patches):

* commit subject: must be less than 50 characters long so something like `ar71xx: Add support for Mikrotik RouterBOARD SXT 2nD r3` should work

* commit message: if you add support for new hardware: Include in your commit message a short description of the hardware and how to install OpenWrt on it. Have a look at the [recent additions](https://git.openwrt.org/?p=openwrt/openwrt.git&a=search&h=HEAD&st=commit&s=add+support+for) for some examples.

ynezz commented on target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt:
```diff
@@ -1176,11 +1176,12 @@ config ATH79_MACH_RBSPI
 
 config ATH79_MACH_RBSXTLITE
-	bool "MikroTik RouterBOARD SXT Lite"
+	bool "MikroTik RouterBOARD SXT Lite (NAND flash versions)"
```
Please don't touch code which is not related to this commit.

ynezz commented on target/linux/ar71xx/base-files/etc/board.d/02_network:
```diff
@@ -113,6 +113,7 @@ ar71xx_setup_interfaces()
 	rb-mapl-2nd|\
 	rb-sxt2n|\
 	rb-sxt5n|\
+	rb-sxt-2nd-r3|\
```
It should be sorted as:

```
rb-sxt2n
rb-sxt-2nd-r3
rb-sxt5n
```

zorxd committed:
ar71xx: Add support for Mikrotik RB SXT 2nD r3

The Mikrotik RouterBOARD SXT 2nD r3 is an outdoor WiFi AP / CPE
with a single 2.4 GHz radio and a 100 Mbps Ethernet port.
The device similar to the SXT 2nD r2, but it has SPI NOR flash instead
of NAND flash.

Hardware
--------
CPU:   Atheros AR9344 (600 MHz)
RAM:   64 MiB
FLASH: 16 MiB SPI NOR W25Q128
ETH:   1x 100 Mbps Atheros AG71xx
WiFi: 2T2R 802.11b/g/n (ath9k)
Power: Passive PoE 8-30 V

Installation instructions:

1. Boot openwrt-ar71xx-mikrotik-vmlinux-initramfs.elf using a
DHCP+TFTP server.
2. Erase the "firmware" partition using the mtd command. This should
no longer be required once this patch is merged.
3. Use sysupgrade to install to flash. The file
openwrt-ar71xx-mikrotik-rb-nor-flash-16M-squashfs-sysupgrade.bin
should be used.

Signed-off-by: Xavier Douville <github@douville.org>

zorxd commented:
Hi
Thanks for your help. I think I addressed your comments in my revised patch.
I would contribute to ath79 as well but it looks like there is no Mikrotik device supported yet so that would require much more work.

Xavier

ynezz commented:
Thanks! I've just merged it to my [staging tree](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=commit;h=f8253e6b19bf122d0d51aa7bd28c5fd2476c89d1)
